modules/extension/wmts/src/main/java/org/geotools/ows/wmts/map/WMTSMapLayer.java

```diff
@@ -99,8 +99,9 @@ public synchronized ReferencedEnvelope getBounds() {
     }
 
     /** Returns the {@link WebMapTileServer} used by this layer */
+    @Deprecated
     public WebMapTileServer getWebMapServer() {
-        return getReader().wmts;
+        throw new UnsupportedOperationException();
```

Comment:
No, this deprecates and removes the functionality at the same time... it's not how deprecation works.
Deprecated methods must keep on working while they are deprecated, then they can be broken right away.

Reply:
I've reentered the old functionality.